setup.py

```diff
@@ -1,136 +1,8 @@
-
-    def read_configuration(filepath):
-        """Read metadata and options from setup.cfg located at filepath."""
-        cfg = ConfigParser()
-        with io.open(filepath, encoding='utf-8') as f:
-            cfg.read_file(f)
-
-        md = dict(cfg.items('metadata'))
-        for list_key in 'classifiers', 'keywords':
-            try:
-                md[list_key] = cfg_val_to_list(md[list_key])
-            except KeyError:
-                pass
-        try:
-            md['long_description'] = maybe_read_files(md['long_description'])
-        except KeyError:
-            pass
-        opt = dict(cfg.items('options'))
-        for list_key in 'use_scm_version', 'zip_safe':
-            try:
-                opt[list_key] = cfg_val_to_primitive(opt[list_key])
-            except KeyError:
-                pass
-        for list_key in 'scripts', 'install_requires', 'setup_requires':
-            try:
-                opt[list_key] = cfg_val_to_list(opt[list_key])
-            except KeyError:
-                pass
-        try:
-            opt['package_dir'] = cfg_val_to_dict(opt['package_dir'])
-        except KeyError:
-            pass
-        opt_package_data = dict(cfg.items('options.package_data'))
-        try:
-            if not opt_package_data.get('', '').strip():
-                opt_package_data[''] = opt_package_data['*']
-                del opt_package_data['*']
-        except KeyError:
-            pass
-        try:
-            opt_extras_require = dict(cfg.items('options.extras_require'))
-            opt['extras_require'] = {}
-            for k, v in opt_extras_require.items():
-                opt['extras_require'][k] = cfg_val_to_list(v)
-        except NoSectionError:
-            pass
-        opt['package_data'] = {}
-        for k, v in opt_package_data.items():
-            opt['package_data'][k] = cfg_val_to_list(v)
-        cur_pkgs = opt.get('packages', '').strip()
-        if '\n' in cur_pkgs:
-            opt['packages'] = cfg_val_to_list(opt['packages'])
-        elif cur_pkgs.startswith('find:'):
-            opt_packages_find = dict(cfg.items('options.packages.find'))
-            opt['packages'] = setuptools.find_packages(**opt_packages_find)
-        return {'metadata': md, 'options': opt}
-
-
-setup_params = {}
-declarative_setup_params = read_configuration('setup.cfg')
-
-# Patch incorrectly decoded package_dir option
-# ``egg_info`` demands native strings failing with unicode under Python 2
-# Ref https://github.com/pypa/setuptools/issues/1136
-if 'package_dir' in declarative_setup_params['options']:
-    declarative_setup_params['options']['package_dir'] = {
-        str(k): str(v)
-        for k, v in declarative_setup_params['options']['package_dir'].items()
-    }
-
-setup_params = dict(setup_params, **declarative_setup_params['metadata'])
-setup_params = dict(setup_params, **declarative_setup_params['options'])
+# Project skeleton maintained at https://github.com/jaraco/skeleton
```

Comment:
I'd like to keep shim up until we drop python 2 here. My goal was to never touch `setup.py` again. Also, it tightens pre-requisites requirements which is unfriendly to users.

Reply:
With wheels published, most users won't be affected by setup.py or its prerequisites at all. Plus, with pip 10, available with most envs, including those defined by this project, the pyproject.toml designates the requirements, so there's no burden on the users.